src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -295,8 +295,12 @@ class WalletService(
             .findByIdAndUserId(walletId.value.toString(), xUserId.id.toString())
             .switchIfEmpty { Mono.error(WalletNotFoundException(walletId)) }
             .map { it.toDomain() }
-            .filter { it.status == WalletStatusDto.CREATED }
-            .switchIfEmpty { Mono.error(WalletConflictStatusException(walletId)) }
+            .flatMap {
+                when (it.status) {
+                    WalletStatusDto.CREATED -> Mono.just(it)
+                    else -> Mono.error(WalletConflictStatusException(walletId, it.status))
+                }
+            }
```

Comment:
what about generalize this replicated code in a function with wanted wallet status and that throw the `WalletConflictStatusException` exception in case of exception mismatch?

Reply:
Yes, I'm totally agree

Reply:
Solved by https://github.com/pagopa/pagopa-wallet-service/pull/150/commits/2043d528e69de69d75a670d36b48a6838475b0b3